simple-backend/nlpviewer_backend/handlers/document.py

```diff
@@ -32,7 +47,21 @@ def create(request):
 
 @require_login
 def edit(request, document_id):
-    doc = Document.objects.get(pk=document_id)
+    """
+    edit a document, query by id.
+    accessible for users with 'change' permission and the owner of the project.
+    """
+    # if doc doen't exist
+    try:
+        doc = Document.objects.get(pk=document_id)
+    except ObjectDoesNotExist: 
+        return HttpResponse('not found', status=404)
+
+    # check permissions
+    if not request.user.has_perm('nlpviewer_backend.change_document') and request.user != doc.project.user:
```

Comment:
So the permission condition here is:

1. The user can change document
2. The user belongs to this project

What happen if we want one user to be able to edit one project but not the other one?